src/models/Likes_DislikesModel.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+declare(strict_types=1);
+
+class LikesModel
+{
+    public PDO $db;
+
+    public function __construct(PDO $db)
+    {
+        $this->db = $db;
+    }
+
+    public function trackActivity(int $postsId, int $usersId): array
+    {
+        $query = $this->db->prepare(
+            'SELECT `posts_id`, `users_id`
+        FROM `likes_dislikes`
+        WHERE `posts_id` = :postsId AND `users_id` = :usersId'
+        );
+        $query->execute([
+            ':postsId' => $postsId,
+            ':usersId' => $usersId
+        ]);
+
+        $result = $query->fetch();
+
+        if (!$result) {
+            $output = [];
+            return $output;
+        } else {
+            return $result;
+        }
+    }
+
+        public function addDislike(int $postsId, int $usersId, array $dislikeCheck): void
+        {
+        if (!$dislikeCheck) {
+            $query = $this->db->prepare('INSERT INTO `likes_dislikes` (`posts_id`, `users_id`)
+                                                VALUES (:postsId, :usersId);');
+            $query->execute([
+                    ':postsId' => $postsId,
+                    ':usersId' => $usersId
+                    ]);
+
```

Comment:
This inserting bit is actually the same in the likes and dislikes function still. It's like an `addActivity` function (I don't like that name, but if getting like/dislikes existing is 'track activity' then this is 'add activity'). 

The only bit where these functions are different is where you add to the dislikes count or the likes count. 

Might there be a data structure that lets you store *who* liked and disliked? Which would give you a way to count likes and dislikes without updating a count on posts? Then removing likes and dislikes later would be easy too.